nengo/utils/least_squares_solvers.py

```diff
@@ -106,8 +121,16 @@ def __call__(self, A, Y, sigma, rng=None):
             def callback(x, i=i):
                 itns[i] += 1
 
-            X[:, i], infos[i] = scipy.sparse.linalg.cg(
-                G, B[:, i], tol=self.tol, callback=callback)
+            try:
+                X[:, i], infos[i] = scipy.sparse.linalg.cg(
+                    G, B[:, i], tol=self.tol, callback=callback,
+                    atol=self.atol)
+            except TypeError as e:
+                # no atol parameter in Scipy < 1.0
+                if 'atol' not in str(e):
+                    raise e
+                X[:, i], infos[i] = scipy.sparse.linalg.cg(
+                    G, B[:, i], tol=self.tol, callback=callback)
```

Comment:
Should `.travis.yml` be modified to test both scipy < 1.0 and the current version? Since here we're explicitly supporting both versions, it would be good to make sure we don't break that in the future.

Reply:
Testing everything with scipy < 1.0 seems like overkill just for this one case. I guess there should be a way to run just this test, so that might be worth it (not sure how much time just spooling up a new VM takes, though).

Frankly, I'm not too worried. I doubt that this would break, since I think it's pretty clear in the code what it's doing, so nobody should change it. But if it did break for scipy < 1.0, that's not a huge deal IMO. Most users should be able to upgrade their scipy pretty easily.

Reply:
My thinking / intent is the check wouldn't be in place just for this code. The check would be for any other code that exists now, or in the future, that might differ across the two versions. If something else breaks across versions, but we ask that people upgrade scipy, then the above code ends up being unnecessary / unused. On the other hand, if we are trying to support scipy < 1.0, then we won't know for sure when new code is added if it's not tested. So I'm mainly looking for the the formal stance about what versions we're looking to support, and positing that for long-term consistency it makes sense to test both until support is dropped. Like you said, it should be easy enough to upgrade scipy if someone wants to use this solver, so we could just drop support now?

Reply:
Might be worth mentioning that the `atol` is present in 1.1.0, but [_not_ 1.0.0](https://docs.scipy.org/doc/scipy-1.0.0/reference/generated/scipy.sparse.linalg.cg.html), so it's actually not a version 1.0.0 difference.